core/trino-main/src/main/java/io/trino/execution/warnings/DefaultWarningCollector.java

```diff
@@ -51,4 +55,12 @@ public synchronized List<TrinoWarning> getWarnings()
     {
         return ImmutableList.copyOf(warnings);
     }
+
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        // do not synchronize on purpose
+        return INSTANCE_SIZE
+                + estimatedSizeOf(warnings, TrinoWarning::getRetainedSizeInBytes);
```

Comment:
if it was OK to do this, you would need to resolve IDE warning

but it's not OK to do this.
when a thread unsafe collection is accessed & modified concurrently, nothing is guaranteed, including no guarantee that collection method return at all

Reply:
BTW error-prone should flag this, but we disabled it